sqlmesh/core/snapshot/definition.py

```diff
@@ -751,6 +727,61 @@ def version_get_or_generate(self) -> str:
         """Helper method to get the version or generate it from the fingerprint."""
         return self.version or self.fingerprint.to_version()
 
+    def is_valid_start(
+        self, start: t.Optional[TimeLike], snapshot_default_start: t.Optional[TimeLike] = None
+    ) -> bool:
+        """Checks if the given start and end are valid for this snapshot.
+
+        Args:
+            start: The start date/time of the interval (inclusive)
+            snapshot_default_start: The default start date/time of the snapshot if one is not defined (inclusive)
+        """
+        # The snapshot may not have a start defined. If so we use the provided default start.
+        snapshot_start = self.start or snapshot_default_start
+        if self.depends_on_past and start:
+            assert snapshot_start, "Snapshot must have a start defined if it depends on past"
+            start_ts = to_timestamp(self.model.cron_floor(start))
+            if not self.intervals and start_ts > to_timestamp(snapshot_start):
```

Comment:
Should this be:
```
if not self.intervals:
    return to_timestamp(snapshot_start) >= start_ts
```
What's the point of proceeding to missing intervals otherwise.